src/sync/sync_user.cpp

```diff
@@ -21,44 +21,123 @@
 #include "sync/impl/sync_metadata.hpp"
 #include "sync/sync_manager.hpp"
 #include "sync/sync_session.hpp"
+#include "sync/generic_network_transport.hpp"
+#include "../../external/json/json.hpp"
+
+#include <realm/util/base64.hpp>
 
 namespace realm {
 
+static std::string base64_decode(const std::string &in) {
+    std::string out;
+    out.resize(util::base64_decoded_size(in.size()));
+    util::base64_decode(in, &out[0], out.size());
+    return out;
+}
+
+static std::vector<std::string> split_token(const std::string& jwt) {
```

Comment:
There don't seem to be any tests for this function? Is it tested implicitly elsewhere because it seems rather complex?

Reply:
It is tested implicitly by using real JWTs in our tests. However, it should have more robust testing.